### DISCUSSION TITLE
NetCDF format options for new point/grid observations

### DISCUSSION THREAD

**Discussion by barlage** [dtcenter/METplus#1401] [File I/O] [answered]
The EMC land team would like to reformat some of our point data (maybe some grid data) into a format that METplus can use (while maintaining their usability in our other scripts). Is there documentation related to NetCDF formatting for new METplus obs data (point and grid)? I can't seem to find any relevant section in the standard docs.

When I talked with @JohnHalleyGotway about this before, I thought he mentioned there was a data convention (maybe CF?) that would work in MET. Or maybe there was an existing wrapper for one of these conventions?

Our hope is that since we are reformatting the data/obs from many native formats to NetCDF, we should use a format that would be compatible with MET for point verification, e.g. flux tower or soil probes (or maybe even grid verification, e.g. GLEAM ET), either now or in the future. This is all part of my dream to use METplus for land evaluation more generally.

**Comment by JohnHalleyGotway** [comment, marked as answer]
Thanks for starting this discussion @barlage. Here's my last response from our previous email chain.

Yes, for gridded data, I recommend that you reformat to the NetCDF [Climate-Forecast convention](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html).

However, as of MET version 10.0.0, it does not support the CF conventions for point data. All of the existing MET point pre-processing tools (ascii2nc, pb2nc, madis2nc, and so on) write to a common intermediate NetCDF point observation format. But I don't recommend that you write to that format. It is just an internal format and not generally supported or well-documented. I do recall thinking that we SHOULD enhance MET to support the CF conventions for point data, but we do not yet. And frankly, that won't happen until a clear need is identified along with funding to pay for it.

Here's one option. You could write to that CF point format described in [this section](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html#point-data). Until we get to the point of MET supporting that directly, we could write a python-embedding script to read those obs into MET.

**Reply by JohnHalleyGotway** [reply]
@barlage, wondering if you are seeking any more input on this discussion. Or should we mark it as being answered? The METplus team monitors open discussions daily, so we try get them wrapped up when possible.

**Reply by barlage** [reply]
@JohnHalleyGotway sure, we can close it. we are currently reformatting several point observation sources so will re-open if we need more feedback (if that's the procedure). thanks!
